include/nbl/builtin/hlsl/tgmath.hlsl

```diff
@@ -4,70 +4,130 @@
+template<typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+inline FloatingPoint erf(FloatingPoint x)
+{
+    return tgmath_impl::erf_helper<FloatingPoint>::__call(x);
+}
+
+template<typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+inline FloatingPoint erfInv(FloatingPoint x)
+{
+    return tgmath_impl::erfInv_helper<FloatingPoint>::__call(x);
+}
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::Vectorial<T>)
+inline T floor(NBL_CONST_REF_ARG(T) val)
+{
+    return tgmath_impl::floor_helper<T>::__call(val);
+}
+
+template<typename T, typename U NBL_FUNC_REQUIRES((concepts::FloatingPointLikeScalar<T> || concepts::FloatingPointLikeVectorial<T>) && (concepts::FloatingPointLikeScalar<U> || is_same_v<U, bool>))
+inline T lerp(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y, NBL_CONST_REF_ARG(U) a)
+{
+    return tgmath_impl::lerp_helper<T, U>::__call(x, y, a);
+}
+
+template<typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+inline bool isnan(NBL_CONST_REF_ARG(FloatingPoint) val)
+{
+    return tgmath_impl::isnan_helper<FloatingPoint>::__call(val);
+}
+
+template<typename V NBL_FUNC_REQUIRES(concepts::FloatingPointLikeVectorial<V>)
+inline vector<bool, vector_traits<V>::Dimension> isnan(NBL_CONST_REF_ARG(V) val)
+{
+    return tgmath_impl::isnan_helper<V>::__call(val);
+}
+
+template<typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+inline FloatingPoint isinf(NBL_CONST_REF_ARG(FloatingPoint) val)
+{
+    return tgmath_impl::isinf_helper<FloatingPoint>::__call(val);
+}
+
+template<typename V NBL_FUNC_REQUIRES(concepts::FloatingPointLikeVectorial<V>)
+inline vector<bool, vector_traits<V>::Dimension> isinf(NBL_CONST_REF_ARG(V) val)
+{
+    return tgmath_impl::isinf_helper<V>::__call(val);
+}
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::FloatingPointLikeVectorial<T>)
+inline T pow(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y)
+{
+    return tgmath_impl::pow_helper<T>::__call(x, y);
+}
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::FloatingPointLikeVectorial<T>)
+inline T exp(NBL_CONST_REF_ARG(T) x)
+{
+    return tgmath_impl::exp_helper<T>::__call(x);
+}
+
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::Vectorial<T>)
+inline T exp2(NBL_CONST_REF_ARG(T) x)
+{
+    return tgmath_impl::exp2_helper<T>::__call(x);
+}
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::FloatingPointLikeVectorial<T>)
+inline T log(NBL_CONST_REF_ARG(T) x)
+{
+    return tgmath_impl::log_helper<T>::__call(x);
+}
+
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<T> || concepts::SignedIntegral<T> || concepts::FloatingPointLikeVectorial<T> || concepts::SignedIntVectorial<T>)
```

Comment:
maybe don't put any constraints on the functions, and let the helpers do all the error reporting (no point doing it twice) as long as each `_helper` declares `NBL_STRUCT_CONSTRAINABLE` in the EMPTY forward declaration of the primary template

Reply:
done here: https://github.com/Devsh-Graphics-Programming/Nabla/pull/817